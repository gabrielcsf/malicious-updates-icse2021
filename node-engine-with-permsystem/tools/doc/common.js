'use strict';

const yaml = require('js-yaml');

function isYAMLBlock(text) {
  return !!text.match(/^<!-- YAML/);
}

exports.isYAMLBlock = isYAMLBlock;

function arrify(value) {
  return Array.isArray(value) ? value : [value];
}

function extractAndParseYAML(text) {
  text = text.trim();

  text = text.replace(/^<!-- YAML/, '')
             .replace(/-->$/, '');

  // js-yaml.safeLoad() throws on error
  const meta = yaml.safeLoad(text);

  const added = meta.added || meta.Added;
  if (added) {
    // Since semver-minors can trickle down to previous major versions,
    // features may have been added in multiple versions.
    meta.added = arrify(added);
  }

  const deprecated = meta.deprecated || meta.Deprecated;
  if (deprecated) {
    // Treat deprecated like added for consistency.
    meta.deprecated = arrify(deprecated);
  }

  return meta;
}

exports.extractAndParseYAML = extractAndParseYAML;
