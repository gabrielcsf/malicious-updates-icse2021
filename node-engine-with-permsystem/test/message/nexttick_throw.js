'use strict';
require('../common');

process.nextTick(function() {
  process.nextTick(function() {
    process.nextTick(function() {
      process.nextTick(function() {
        // eslint-disable-next-line
        undefined_reference_error_maker;
      });
    });
  });
});
