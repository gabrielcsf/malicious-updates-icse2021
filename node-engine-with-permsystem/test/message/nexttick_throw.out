
*test*message*nexttick_throw.js:*
        undefined_reference_error_maker;
        ^
ReferenceError: undefined_reference_error_maker is not defined
    at *test*message*nexttick_throw.js:*:*
    at _combinedTickCallback (internal/process/next_tick.js:*:*)
    at process._tickCallback (internal/process/next_tick.js:*:*)
    at Module.runMain (module.js:*:*)
    at run (bootstrap_node.js:*:*)
    at startup (bootstrap_node.js:*:*)
    at bootstrap_node.js:*:*
